scripts/agent_step_zkai_stwo_composition.py

```
@@ -0,0 +1,588 @@
+        raise CompositionError("Stwo statement-envelope mutation count is incomplete")
+    if proof_only.get("mutations_rejected") != 1:
+        raise CompositionError("Stwo proof-only calibration no longer matches the checked gate")
+    return payload
+
+
+def _checked_stwo_payload(stwo_evidence: dict[str, Any]) -> dict[str, Any]:
+    checked_stwo_evidence_path = stwo_evidence.get("_source_path")
+    if checked_stwo_evidence_path is not None:
+        return checked_stwo_evidence(pathlib.Path(checked_stwo_evidence_path))
+    if not STWO.benchmark_passed(stwo_evidence):
+        raise CompositionError("Stwo benchmark payload no longer passes")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)